### ISSUE TITLE
Protractor - throw er; // Unhandled 'error' event

### ISSUE THREAD

**Issue by mnemanja** [jakemmarsh/angularjs-gulp-browserify-boilerplate#160]
Simply running gulp protractor provides this error:

[14:16:06] Finished 'gzip' after 205 ms
[14:16:06] Finished 'prod' after 4.37 s
[14:16:06] Starting 'protractor'...
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::3000
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at Server._listen2 (net.js:1234:14)
    at listen (net.js:1270:10)
    at Server.listen (net.js:1366:5)
    at EventEmitter.listen (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\express\lib\application.js:617:24)
    at testServer.js:12:24
    at new Promise (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\karma\node_modules\core-js\modules\es6.promise.js:186:7)
    at testServer (testServer.js:11:10)
    at Gulp.<anonymous> (protractor.js:18:3)
    at module.exports (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\lib\runTask.js:34:7)
    at Gulp.Orchestrator._runTask (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\index.js:273:3)
    at Gulp.Orchestrator._runStep (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\index.js:214:10)
    at C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\index.js:279:18
    at finish (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\lib\runTask.js:21:8)
    at cb (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp\node_modules\orchestrator\lib\runTask.js:29:3)

I've tested this on the boilerplate and in the project, produces the same error.

**Comment by mnemanja** [comment]
Never mind, I missed this updated in the **gulp/config.js**
`assetExtensions: [`

All is good. GJ.

**Comment by jakemmarsh** [comment]
Are you running `gulp dev` (or some other server) at the same time? The first line of the error, `Error: listen EADDRINUSE :::3000`, means that something is already being run at port 3000.

**Comment by mnemanja** [comment]
Today I've done the following:
gulp protractor - fail with the error **[1]**
gulp dev + gulp protractor - fail with the old error **[2]**
just gulp protractor again - fail with the error **[1]**

I've tried `npm install` and everything went well: `selenium standalone is up to date.` but the `no selenium server` error remains. 
All of this was done on the clean boilerplate.


Errors:
**[1]**
```
09:23:17] Starting 'gzip'...
[09:23:18] Finished 'gzip' after 205 ms
[09:23:18] Finished 'prod' after 7.02 s
[09:23:18] Starting 'protractor'...
[launcher] Process exited with error code 1
C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp-protractor\node_m                                                                                                                                                                                               odules\protractor\node_modules\q\q.js:126
                    throw e;
                    ^

Error: No selenium server jar found at the specified location (C:\www\angularjs-                                                                                                                                                                                               gulp-browserify-boilerplate\node_modules\gulp-protractor\node_modules\protractor                                                                                                                                                                                               \selenium\selenium-server-standalone-2.48.2.jar). Check that the version number                                                                                                                                                                                                is up to date.
    at [object Object].LocalDriverProvider.addDefaultBinaryLocs_ (C:\www\angular                                                                                                                                                                                               js-gulp-browserify-boilerplate\node_modules\gulp-protractor\node_modules\protrac                                                                                                                                                                                               tor\lib\driverProviders\local.js:37:11)
    at [object Object].LocalDriverProvider.setupEnv (C:\www\angularjs-gulp-brows                                                                                                                                                                                               erify-boilerplate\node_modules\gulp-protractor\node_modules\protractor\lib\drive                                                                                                                                                                                               rProviders\local.js:71:8)
    at [object Object].Runner.run (C:\www\angularjs-gulp-browserify-boilerplate\                                                                                                                                                                                               node_modules\gulp-protractor\node_modules\protractor\lib\runner.js:274:31)
    at [object Object].TaskRunner.run (C:\www\angularjs-gulp-browserify-boilerpl                                                                                                                                                                                               ate\node_modules\gulp-protractor\node_modules\protractor\lib\taskRunner.js:123:1                                                                                                                                                                                               9)
    at createNextTaskRunner (C:\www\angularjs-gulp-browserify-boilerplate\node_m                                                                                                                                                                                               odules\gulp-protractor\node_modules\protractor\lib\launcher.js:216:20)
    at C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp-protractor                                                                                                                                                                                               \node_modules\protractor\lib\launcher.js:239:7
    at _fulfilled (C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gul                                                                                                                                                                                               p-protractor\node_modules\protractor\node_modules\q\q.js:797:54)
    at self.promiseDispatch.done (C:\www\angularjs-gulp-browserify-boilerplate\n                                                                                                                                                                                               ode_modules\gulp-protractor\node_modules\protractor\node_modules\q\q.js:826:30)
    at Promise.promise.promiseDispatch (C:\www\angularjs-gulp-browserify-boilerp                                                                                                                                                                                               late\node_modules\gulp-protractor\node_modules\protractor\node_modules\q\q.js:75                                                                                                                                                                                               9:13)
    at C:\www\angularjs-gulp-browserify-boilerplate\node_modules\gulp-protractor                                                                                                                                                                                               \node_modules\protractor\node_modules\q\q.js:573:44

C:\www\angularjs-gulp-browserify-boilerplate\gulp\tasks\protractor.js:37
      throw err;
      ^
Error: protractor exited with code 1

```

**[2]**
```
Error: listen EADDRINUSE :::3000
```

**Comment by saesh** [comment]
@mnemanja You need to update the webdriver using the `gulp-protractor` dependant `protractor`: `./node_modules/gulp-protractor/node_modules/protractor/bin/webdriver-manager update`

**Comment by mnemanja** [comment]
Shouldn't it already have the latest update, since it is the fresh install of the boilerplate and node modules?

**Comment by saesh** [comment]
@mnemanja In the `package.json` the script `webdriver-update` updates the webdriver of the `protractor` dependency but not the webdriver of `gulp-protractor`, which also has a `protractor` dependency itself and uses this one to run the tests when you run `gulp test`.

**Comment by mnemanja** [comment]
@saesh Thanks for the assistance. Maybe I'm not getting exactly what needs to be done, since the way I've applied it didn't help.
This is the current state of my "scripts" in the package.json

```
  "scripts": {
    "test": "./node_modules/.bin/gulp test",
    "webdriver-update": "./node_modules/.bin/webdriver-manager update",
    "protractor": "./node_modules/gulp-protractor/node_modules/protractor/bin/webdriver-manager update",
    "postinstall": "npm run webdriver-update"
  },
```

**Comment by saesh** [comment]
@mnemanja Try this:

```
 "scripts": {
    "test": "./node_modules/.bin/gulp test",
    "webdriver-update": "./node_modules/gulp-protractor/node_modules/protractor/bin/webdriver-manager update",
    "postinstall": "npm run webdriver-update"
  },
```

And then run `gulp test` or `gulp protractor`.

**Comment by mnemanja** [comment]
Thanks @saesh. Works :)

@jakemmarsh Previously I believe I was able to run a local development and the protractor at the same time, same goes for karma and local development. The port would just get incremented (3000, 3001, 3002, 3003...).

This currently means that I can't run my E2E tests - protractor locally. Would we consider this a bug, or a new :( feature?

**Comment by jakemmarsh** [comment]
Hm, I don't remember purposely making any changes like that. Easy enough fix though, see #170
